### DISCUSSION TITLE
Local development keys and Docker

### DISCUSSION THREAD

**Discussion by colvin** [envkey/envkey#45] [Q&A] [answered]
I really like the v2 feature where local development keys are automatically generated, but I have a couple of circumstances where I'm not sure how to best handle things.

I have a project (that I'm transitioning from v1 to v2) that involves some activity running directly on my machine and some activity that occurs within a container. For the direct activity my auto-generated development key will work great, but for the containerized activity I'll need to propagate the ENVKEY into the container via a `.env` file.

Should I locate the auto-generated key in `$HOME/.envkey/apps/` and move it to a `.env` file? Should I copy it maybe? Generating a separate local development key seems a bit overkill. What is the best advice here?

I also can't see a way to view the app ID from within the UI. I can go and look at the contents of the `.envkey` file I generated with `envkey init`, but it would be nice to obtain it from the UI as well. I used the ID to find the correct auto-generated env file.

**Comment by danenania** [comment, marked as answer]
Hey @colvin, I think any of the options you mention could potentially work: forwarding the ENVKEY to the container as an env var, copying the file over from `~/.envkey/apps`, or generating a separate local development ENVKEY.

Off the top of my head, I can't think of a particularly strong reason to prefer one option over the others, but I think I'd probably go with copying over the `~/.envkey/apps` file into the container.

I agree that showing the app id in the UI would be helpful for this situation. In the meantime, you can also get it from the CLI using `envkey apps --json-pretty`

**Reply by colvin** [reply]
I appreciate the quick response! Since we're upgrading to v2 I'm writing some docs for the other developers in our org, and for the projects that require a `.env` file I wanted to document a preferred approach to setting that up. I think I may just recommend for those apps that users go ahead and generate a local development key for themselves at the outset just to make it simple.

If one generates a local development key before using the app for the first time, does a one still get auto-generated?

**Reply by danenania** [reply]
That sounds like a good approach--it will be the most similar to the v1 workflow, so it should help to keep the transition simple.

> If one generates a local development key before using the app for the first time, does a one still get auto-generated?

If an `ENVKEY` is already set, one won't be auto-generated. A local development ENVKEY will only be auto-generated if the `ENVKEY` is missing and the EnvKey CLI is available in the `$PATH`.

**Reply by colvin** [reply]
Thanks!
